machine/machine.go

```diff
@@ -313,3 +313,57 @@ func getNumCores() int {
 
 	return numCPU
 }
+
+var cpuToSocket map[int]int
+
+// GetSocketFromCPU returns socket which is bounded to passed CPU.
+// If there is no information it would return -1.
+func GetSocketFromCPU(cpu int) int {
+	// Check if information is cached.
+	if cpuToSocket == nil {
+		var err error
+		cpuToSocket, err = readCPUToSocket(cpuBusPath)
+		if err != nil {
+			klog.Errorf("Couldn't gather information about CPU bounded to socket: %v", err)
+		}
+	}
+
+	if _, ok := cpuToSocket[cpu]; !ok {
+		return -1
+	}
+
+	return cpuToSocket[cpu]
+}
+
+func readCPUToSocket(basePath string) (map[int]int, error) {
+	readData := make(map[int]int)
+
+	pathPattern := basePath + "cpu*[0-9]"
+	sysCPUPaths, err := filepath.Glob(pathPattern)
+	if err != nil {
+		return nil, err
+	}
+	cpuToSocket = make(map[int]int)
+	for _, sysCPUPath := range sysCPUPaths {
+		propertyPath := filepath.Join(sysCPUPath, sysFsCPUTopology, sysFsCPUPhysicalPackageID)
+		propertyVal, err := ioutil.ReadFile(propertyPath)
+		if err != nil {
+			return nil, err
+		}
+
+		cpuDir := path.Base(sysCPUPath)
+
+		// Get CPU number from directory name ( ex. cpu10 ).
+		cpu, err := strconv.ParseInt(cpuDir[3:], 10, 0)
```

Comment:
Is it possible that `cpuDir` has length equal to 3 or less? Consider length checking before accessing the third element of `cpuDir`.